joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeHintCallLinker.scala

```diff
@@ -104,16 +105,58 @@ abstract class XTypeHintCallLinker(cpg: Cpg) extends CpgPass(cpg) {
       if (methodName.contains(pathSep) && methodName.length > methodName.lastIndexOf(pathSep) + 1)
         methodName.substring(methodName.lastIndexOf(pathSep) + 1)
       else methodName
-    MethodStubCreator
+    val stub = MethodStubCreator
       .createMethodStub(
         name,
         methodName,
         "",
         DispatchTypes.DYNAMIC_DISPATCH.name(),
         call.argumentOut.size,
         builder,
-        isExternal
+        isExternal,
+        astParentFullName = XTypeHintCallLinker.namespace
       )
+    addTypeDeclInfo(stub, builder)
   }
 
+  /** Try to extract a type full name from the method full name, if one exists in the CPG then we are lucky and we use
+    * it, else we ignore for now.
+    */
+  protected def addTypeDeclInfo(stub: NewMethod, builder: DiffGraphBuilder): NewMethod = {
+    val nameIdx = stub.fullName.lastIndexOf(stub.name)
+    if (!stub.fullName.isBlank && !stub.fullName.startsWith("<operator") && nameIdx > 0) {
+      cpg.typeDecl.fullNameExact(stub.fullName.substring(0, nameIdx - 1)).headOption.foreach { typeDecl =>
+        stub
+          .astParentFullName(typeDecl.fullName)
+          .astParentType(NodeTypes.TYPE_DECL)
+        builder.addEdge(typeDecl, stub, EdgeTypes.AST)
+      }
+      stub
+    } else {
+      stub
+    }
+  }
```

Comment:
I would recommend rewriting this to a method which calculated `astParentFullName` and `astParentType` not based on `stub: NewMethod` but rather `fullName: Stirng`. That way we can directly call `createMethodStub` with the approriate values and dont need to overwrite the after stub creation.

Reply:
@ml86 made the adjustment